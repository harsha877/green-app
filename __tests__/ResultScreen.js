import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import ResultScreen from '../src/views/resultsScreen';

const push = jest.fn();
const route={params:{result:90,isAccesibilityModeOn:false}}

test("result Screen Loading",()=>{
   render(< ResultScreen route={route} navigation={{ push }}/>)

})

test("result Screen snapshot",()=>{
   let resultComponenet=render(<ResultScreen route={route} navigation={{ push }}/>).toJSON();
   expect(resultComponenet).toMatchSnapshot();
})
describe(" result Screen QueryTesting",()=>{
    let resultComponent;
    beforeEach(()=>{
    resultComponent=render(<ResultScreen route={route} navigation={{ push }}/>);
    })

    
     test("Testing Text",()=>{
       const {getAllByText}=resultComponent;
        text="Congrats you Scored 90%";
        expect(getAllByText(text).length).toBe(1);
     })

     test("Testing Text",()=>{
      const {getAllByText}=resultComponent;
       text="Home";
       expect(getAllByText(text).length).toBe(1);
    })
     
     
     test("Testing Styles",()=>{
        const props=resultComponent.toJSON();
        let Styles=props.props.style;
        expect(Styles).toStrictEqual({
         flex: 1,
         justifyContent: "center",
         backgroundColor: "#063f5c",
         padding: 2,
        });
     })

     test("Testing Styles",()=>{
      const props=resultComponent.toJSON();
      let Styles=props.children[1].props.style;
      expect(Styles).toStrictEqual({
         color: "#ffffff",
         fontSize: 21,
         margin: 5,
         textAlign: "center",
      });
   })
     
});
