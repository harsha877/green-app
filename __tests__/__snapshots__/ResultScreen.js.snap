// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`result Screen snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#063f5c",
      "flex": 1,
      "justifyContent": "center",
      "padding": 2,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#1ca3ec",
          "borderRadius": 50,
          "margin": 10,
          "padding": 0,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
          "width": 125,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 18,
            "padding": 5,
            "textAlign": "center",
          }
        }
      >
        Home
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 21,
        "margin": 5,
        "textAlign": "center",
      }
    }
  >
    Congrats you Scored 
    90
    %
  </Text>
</View>
`;
